src/main/java/online/partyrun/partyrunmatchingservice/global/sse/MultiSinkHandler.java

```diff
@@ -0,0 +1,95 @@
+package online.partyrun.partyrunmatchingservice.global.sse;
+
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+
+import online.partyrun.partyrunmatchingservice.global.sse.exception.KeyNotExistException;
+import online.partyrun.partyrunmatchingservice.global.sse.exception.NullKeyException;
+import online.partyrun.partyrunmatchingservice.global.sse.exception.SseConnectionException;
+
+import org.springframework.stereotype.Component;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Sinks;
+import reactor.core.scheduler.Schedulers;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+@Component
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+public class MultiSinkHandler<K, V> implements ServerSentEventHandler<K, V> {
+    private static final int DEFAULT_MINUTE = 3;
+    Map<K, Sinks.Many<V>> sinks = new HashMap<>();
+
+    @Override
+    public Flux<V> connect(final K key) {
+        return getSink(key)
+                .asFlux()
+                .subscribeOn(Schedulers.boundedElastic())
+                .timeout(timeout())
+                .doOnCancel(() -> sinks.remove(key))
+                .doOnError(
+                        e -> {
+                            sinks.remove(key);
+                            throw new SseConnectionException();
+                        });
+    }
+
+    @Override
+    public void sendEvent(final K key, final V value) {
+        getSink(key).tryEmitNext(value);
+    }
+
+    @Override
+    public void complete(final K key) {
+        getSink(key).tryEmitComplete();
+        sinks.remove(key);
+    }
+
+    @Override
+    public void create(K key) {
+        checkKeyNotNull(key);
+        sinks.putIfAbsent(key, Sinks.many().unicast().onBackpressureBuffer());
+    }
+
+    private Sinks.Many<V> getSink(K key) {
+        validateKey(key);
+        return sinks.get(key);
+    }
+
+    private void validateKey(K key) {
+        checkKeyNotNull(key);
+        if (isNonExists(key)) {
+            throw new KeyNotExistException();
+        }
```

Comment:
메서드로 빼면 좋을 듯하네요.
그리고 들어온 key를 KeyNotExistException 내부로 보내서 메시지로 보여주는게 좋을 것 같네요.
```suggestion
        if (isNonExists(key)) {
            throw new KeyNotExistException(key);
        }
```

Reply:
반영했습니다